src/Compilers/CSharp/Portable/Symbols/FileIdentifier.cs

```diff
@@ -89,11 +89,11 @@ public ImmutableArray<byte> FilePathChecksumOpt
         }
     }
 
-    public static FileIdentifier Create(SyntaxTree tree)
-        => Create(tree.FilePath);
+    public static FileIdentifier Create(SyntaxTree syntaxTree, SourceReferenceResolver? resolver)
+        => new FileIdentifier(syntaxTree.GetNormalizedPath(resolver));
 
-    public static FileIdentifier Create(string filePath)
-        => new FileIdentifier(filePath);
+    public static FileIdentifier Create(string mappedFilePath)
```

Comment:
>mappedFilePath [](http://example.com/codeflow?start=47&length=14)

Perhaps `normalizedPath` would be better to avoid association with #line mapping .

Reply:
Actually, can we remove the overload to avoid mistakes?

Reply:
There is one call I don't fully understand the context of so will need @RikkiGibson to say if there is a way to avoid it

https://github.com/dotnet/roslyn/blob/main/src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs#L76-L80

Reply:
Sorry lost track of this. Digging in to this shortly.

Reply:
It's not the prettiest to leave this internal API here just for EE, but I think it makes EE the most straightforward. The DocumentName comes from the PDB (afaik?) and is already normalized.

We might be able to dig around and get the `SyntaxTree` and `SourceReferenceResolver` in this context but what we have in the PR currently is already correct, so I wouldn't want to spend much time on it.

We could consider renaming the API to indicate that "hey, your file path had better already be normalized when you call this".